File: java/array-1/commonEnd.java
/* Given 2 arrays of ints, a and b, return true if they have the same first 
 * element or they have the same last element. Both arrays will be length 1 or 
 * more.
 */
public boolean commonEnd(int[] a, int[] b) {
    return a[0] == b[0] || a[a.length - 1] == b[b.length - 1];
}

File: python/list-1/common_end.py
# Given 2 arrays of ints, a and b, return True if they have the same first 
# element or they have the same last element. Both arrays will be length 1 or 
# more.
def common_end(a, b):
    return a[0] == b[0] or a[-1] == b[-1]
